packages/plugin-interactive-tools/sources/commands/upgrade-interactive.tsx

```diff
@@ -200,18 +232,41 @@ export default class UpgradeInteractiveCommand extends BaseCommand {
       ]);
       const packageIdentifier = structUtils.stringifyIdent(descriptor);
       const padLength = Math.max(0, 45 - packageIdentifier.length);
-      return <Box>
-        <Box width={45}>
-          <Text bold>
-            {structUtils.prettyIdent(configuration, descriptor)}
-          </Text>
-          <Pad active={active} length={padLength}/>
-        </Box>
-        {suggestions !== null
-          ? <ItemOptions active={active} options={suggestions} value={action} skewer={true} onChange={setAction} sizes={[17, 17, 17]} />
-          : <Box marginLeft={2}><Text color="gray">Fetching suggestions...</Text></Box>
-        }
-      </Box>;
+
+      return (suggestions !== null)
+        ? <UpgradeEntryWithSuggestions active={active} descriptor={descriptor} suggestions={suggestions}/>
+        : <Box>
+          <Box width={45}>
+            <Text bold>
+              {structUtils.prettyIdent(configuration, descriptor)}
+            </Text>
+            <Pad active={active} length={padLength}/>
+          </Box>
+          <Box marginLeft={2}><Text color="gray">Fetching suggestions...</Text></Box>
+        </Box>;
+    };
+
+    const ShowNewUpgrades = ({dependencies}: { dependencies: Array<Descriptor> }) => {
+      const [suggestions, setSuggestions] = useState<Array<[Descriptor, UpgradeSuggestions]>>([]);
+
+      useEffect(() => {
+        Promise.all(dependencies.map(descriptor => fetchSuggestions(descriptor)))
+          .then(allSuggestions => {
+            const mappedToSuggestions = dependencies.map((descriptor, i) => {
+              const suggestionsForDescriptor = allSuggestions[i];
+              return [descriptor, suggestionsForDescriptor] as [Descriptor, UpgradeSuggestions];
```

Comment:
This `as` seems superfluous, but without it the typing of the tuple gets list and the typechecker starts to complain as it seems to think that what's returned with be _either_ a Descriptor _or_ an UpgradeSuggestions, not a tuple of both.

Reply:
Try using `as const`, it usually solve those inference problems

Reply:
Sadly not, but I get a different error! 

Without any `as`, the call to `setSuggestions` has this error:

```
Argument of type '(Descriptor | { value: string | null; label: string; }[])[][]' is not assignable to parameter of type 'SetStateAction<[Descriptor, UpgradeSuggestions][]>'.
```

With `as const`, it's:

```
Argument of type '(readonly [Descriptor, { value: string | null; label: string; }[]])[]' is not assignable to parameter of type 'SetStateAction<[Descriptor, UpgradeSuggestions][]>'.
```

If I set the return type of `fetchSuggestions` explicitly, it just changes the error to substitute `UpgradeSuggestions` instead of the raw type.

Reply:
I believe it would work with `as const` and

```ts
useState<Array<readonly [Descriptor, UpgradeSuggestions]>|null>(null)
```

Reply:
That sorted it, thanks!